hack/verify.sh

```diff
@@ -17,128 +17,140 @@
 # Copyright 2022 The Kepler Contributors
 #
 
-set -e
-set -o pipefail
-set -x
-
-source ./hack/common.sh
-
-CLUSTER_PROVIDER=${CLUSTER_PROVIDER:-kubernetes}
-MANIFESTS_OUT_DIR=${MANIFESTS_OUT_DIR:-"_output/generated-manifest"}
-ARTIFACT_DIR="${ARTIFACT_DIR:-/tmp/artifacts}"
-
-function must_gather() {
-    mkdir -p $ARTIFACT_DIR
-    kubectl describe nodes > "$ARTIFACT_DIR/nodes"
-    kubectl get pods --all-namespaces > "$ARTIFACT_DIR/pods"
-    kubectl get deployment --all-namespaces > "$ARTIFACT_DIR/deployments"
-    kubectl get daemonsets --all-namespaces > "$ARTIFACT_DIR/daemonsets"
-    kubectl get services --all-namespaces > "$ARTIFACT_DIR/services"
-    kubectl get endpoints --all-namespaces > "$ARTIFACT_DIR/endpoints"
-    kubectl describe daemonset kepler-exporter -n kepler > "$ARTIFACT_DIR/kepler-daemonset-describe"
-    kubectl get pods -n kepler -o yaml > "$ARTIFACT_DIR/kepler-pod-yaml"
-    kubectl logs $(kubectl -n kepler get pods -o name) -n kepler > "$ARTIFACT_DIR/kepler-pod-logs"
+set -eu -o pipefail
+
+PROJECT_ROOT="$(git rev-parse --show-toplevel)"
+
+declare -r PROJECT_ROOT
+declare -r MANIFESTS_OUT_DIR="${MANIFESTS_OUT_DIR:-_output/generated-manifest}"
+declare -r EXPORTER="${EXPORTER:-kepler-exporter}"
+declare -r KEPLER_NS="${KEPLER_NS:-kepler}"
+declare -r MONITORING_NS="${MONITORING_NS:-monitoring}"
+
+source "$PROJECT_ROOT/hack/utils.bash"
+
+declare CLUSTER_PROVIDER="${CLUSTER_PROVIDER:-kind}"
+declare CTR_CMD="${CTR_CMD:-docker}"
+declare ARTIFACT_DIR="${ARTIFACT_DIR:-/tmp/artifacts}"
+
+must_gather() {
+	header "Running must gather"
+
+	run kubectl describe nodes | tee "$ARTIFACT_DIR/nodes"
+	run kubectl get pods --all-namespaces | tee "$ARTIFACT_DIR/pods"
+	run kubectl get deployment --all-namespaces | tee "$ARTIFACT_DIR/deployments"
+	run kubectl get daemonsets --all-namespaces | tee "$ARTIFACT_DIR/daemonsets"
+	run kubectl get services --all-namespaces | tee "$ARTIFACT_DIR/services"
+	run kubectl get endpoints --all-namespaces | tee "$ARTIFACT_DIR/endpoints"
+	run kubectl describe daemonset "$EXPORTER" -n "$KEPLER_NS" | tee "$ARTIFACT_DIR/kepler-daemonset-describe"
+	run kubectl get pods -n "$KEPLER_NS" -o yaml | tee "$ARTIFACT_DIR/kepler-pod.yaml"
+	run kubectl logs -n "$KEPLER_NS" daemonset/"$EXPORTER" | tee "$ARTIFACT_DIR/kepler.log"
```

Comment:
suggest to call `log_kepler()` here, and the output file name should be `kepler.log`, not `kepler.logs`